daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala

```diff
@@ -207,13 +203,29 @@ object SValue {
   }
 
   final case class SAny(ty: Type, value: SValue) extends SValue
-  sealed abstract class SException extends SValue
-  final case class SAnyException(ty: Type, value: SValue) extends SException
-  final case class SArithmeticError(
-      builtinName: String,
-      args: ImmArray[String],
-  ) extends SException {
-    def message = s"ArithmeticError while evaluating ($builtinName ${args.iterator.mkString(" ")})."
+  final case class SAnyException(ty: Type, value: SValue) extends SValue
+
+  object SArithmeticError {
+    val tyCon: Ref.TypeConName = Ref.Identifier.assertFromString(
+      "f1cf1ff41057ce327248684089b106d0a1f27c2f092d30f663c919addf173981:DA.Exception.ArithmeticError:ArithmeticError"
```

Comment:
maybe add a comment that this should be kept in sync with the generator or the tests.